### ISSUE TITLE
Shouldn't the default visit option still be `advance`?

### ISSUE THREAD

**Issue by dsander** [turbolinks/turbolinks-rails#10]
I noticed the default visit option changed from [advance](https://github.com/turbolinks/turbolinks-classic/blob/2-5-stable/lib/turbolinks/redirection.rb#L11) to [replace](https://github.com/turbolinks/turbolinks-rails/blob/master/lib/turbolinks/redirection.rb#L26), I don't understand in which use case that would be desirable.

To my understanding [visit_location_with_turbolinks](https://github.com/turbolinks/turbolinks-rails/blob/master/lib/turbolinks/redirection.rb#L24) is only executed for [xhr](https://github.com/turbolinks/turbolinks-rails/blob/master/lib/turbolinks/redirection.rb#L13) requests which aren't GETs , thus never triggered by [Turbolinks.visit](https://github.com/turbolinks/turbolinks/blob/d3b75ae739ddff5e4fcf159d60d3cd84dfa711d8/src/turbolinks/http_request.coffee#L64).
That means that calling `Turbolinks.visit(location, "replace")` will override the history location of the XHR request with the redirection target and navigating back will skip the page that issued the XHR request.
I would expect the default to still be `advance`, because the location of the XHR call is never put in the browser location history.

**Comment by benoitongit** [comment]
I had the same expectation, was surprised when I saw that the default was 'replace'

**Comment by Fudoshiki** [comment]
@rafaelfranca, @sstephenson, @guilleiguaran can your check this please

**Comment by artemave** [comment]
Defaulting to "replace" breaks back button on form submit. Instead of going back to the form - local form behavior - it goes to the page before the form.

**Comment by timdiggins** [comment]
Agreed. For future reference, I got around this by wrapping `redirect_to` in my ApplicationController as follows:

```ruby
  # override Turbolinks.redirect_to to make "advance" the default
  def redirect_to(url = {}, options = {})
    options[:turbolinks] = "advance" unless options.key?(:turbolinks)
    super(url, options)
  end
```

It feels like it would be useful to make this configurable, maybe with new installs having the new default (just switching the default might be too much of a change). However that sounds like it would need to have a proper test suite for this. Are any maintainers/contributors around to give a point of view?
